### PR TITLE
Issue 7: ETL hard delete logic

### PR DESCRIPTION
**Pull request by DangMinhHao**
ETL data of stock table by hard delete pattern

### CODE REVIEW

File: etl/stock_hard_dlt.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+file_upload_dir =  '/etl_' + n_1_date + '.csv'
+
+# Upload to s3 bucket
+s3 = boto3.client('s3',aws_access_key_id=os.environ[], aws_secret_access_key=os.environ[])
+bucket_name = 'analytics-ninjas'
+s3.put_object(Body=csv_buffer.getvalue(), Bucket=bucket_name, Key=file_upload_dir.lstrip('/'))
+
+# Closing allocate memory, and database connection
+csv_buffer.close()
+conn.close()
```
Comment (DangMinhHao) [inline etl/stock_hard_dlt.py:44]: Snapshot data of n-1 day from mysql and upload into s3

File: etl/stock_hard_dlt.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+db_url = os.environ["DB_URL"]
+engine = create_engine(db_url)
+conn = engine.connect()
+
+snapshot = '''
+                SELECT 
+                    *
+                FROM stock
+           '''
```
Comment (boushphong) [inline etl/stock_hard_dlt.py:16]: Minify this into 1 line.

File: etl/stock_hard_dlt.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+snapshot_data = conn.execute(text(snapshot)).fetchall()
+
+# Get column name
+col = conn.execute(text("SHOW COLUMNS FROM ENGINES")).fetchall()
```
Comment (boushphong) [inline etl/stock_hard_dlt.py:20]: Rename Engines please

File: etl/stock_hard_dlt.py (hunk 1 above)
Comment (boushphong) [reply etl/stock_hard_dlt.py:44]: Please end the file here

File: etl/stock_hard_dlt.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+s3.put_object(Body=csv_buffer.getvalue(), Bucket=bucket_name, Key=file_upload_dir.lstrip('/'))
+
+# Closing allocate memory, and database connection
+csv_buffer.close()
+conn.close()
+
+
+
+# Calculate the n-2 day's date
+n_2_date = str(current_date - timedelta(days=2)).replace('-','')
```
Comment (boushphong) [inline etl/stock_hard_dlt.py:49]: Split the file from line 47 to a separate file. and download both `n-1` and `n-2` files to find updated records and put them into the DWH.

**Review by boushphong — APPROVED** [review]

**Comment by boushphong** [comment]
Format with black.

**Pull request closed without merging**